### PR TITLE
MISUV-3589 Create Credits template and controller

### PR DESCRIPTION
**Pull request by andriikondiuk**

### CODE REVIEW

File: app/controllers/CreditsSummaryController.scala [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                                         val agentItvcErrorHandler: AgentItvcErrorHandler
+                                        ) extends ClientConfirmedController with FeatureSwitching with I18nSupport {
+
+  private def getFinancialsByTaxYear(calendarYear: Int, isAgent: Boolean)(f: List[DocumentDetail] => Future[Result])
+                                    (implicit user: MtdItUser[AnyContent]): Future[Result] = {
+    financialDetailsService.getFinancialDetails(calendarYear, user.nino).flatMap {
+      case FinancialDetailsModel(_, _, documentDetails, _) =>
+        val docDetailsMFACredits = documentDetails.filter(_.validMFACreditDescription())
+
+        val financialDetailsWithCredit: List[DocumentDetail] = docDetailsMFACredits.filter { dd =>
```
Comment (joesheridan) [inline app/controllers/CreditsSummaryController.scala:59]: I guess we don't need audit events for this page??

File: app/controllers/CreditsSummaryController.scala [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                                         mcc: MessagesControllerComponents,
+                                         val ec: ExecutionContext,
+                                         val agentItvcErrorHandler: AgentItvcErrorHandler
+                                        ) extends ClientConfirmedController with FeatureSwitching with I18nSupport {
+
+  private def getFinancialsByTaxYear(calendarYear: Int, isAgent: Boolean)(f: List[DocumentDetail] => Future[Result])
+                                    (implicit user: MtdItUser[AnyContent]): Future[Result] = {
+    financialDetailsService.getFinancialDetails(calendarYear, user.nino).flatMap {
+      case FinancialDetailsModel(_, _, documentDetails, _) =>
+        val docDetailsMFACredits = documentDetails.filter(_.validMFACreditDescription())
```
Comment (joesheridan) [inline app/controllers/CreditsSummaryController.scala:57]: are we sure we aren't displaying cutover credits too?

File: app/controllers/CreditsSummaryController.scala [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                                         authenticate: AuthenticationPredicate,
+                                         retrieveNino: NinoPredicate,
+                                         retrieveIncomeSources: IncomeSourceDetailsPredicate)
+                                        (implicit val appConfig: FrontendAppConfig,
+                                         mcc: MessagesControllerComponents,
+                                         val ec: ExecutionContext,
+                                         val agentItvcErrorHandler: AgentItvcErrorHandler
+                                        ) extends ClientConfirmedController with FeatureSwitching with I18nSupport {
+
+  private def getFinancialsByTaxYear(calendarYear: Int, isAgent: Boolean)(f: List[DocumentDetail] => Future[Result])
```
Comment (joesheridan) [inline app/controllers/CreditsSummaryController.scala:53]: f renamed to 'callback' perhaps?

File: app/controllers/CreditsSummaryController.scala (hunk 1 above)
Comment (andriikondiuk) [reply app/controllers/CreditsSummaryController.scala:59]: Yes, I think we do not need the audit events here

**Pull request merged**